### PR TITLE
changed boosted the perfomance for this plugin

### PR DESCRIPTION
**Pull request by mbrueschke**
@plentymarkets/team-multichannel-koboldmaki

### CODE REVIEW

File: src/ResultField/PreisRoboterDE.php [hunk 1]
```diff
@@ -122,24 +140,18 @@ public function generateResultFields(array $formatSettings = []):array
...
-                //attributes
-                'attributes.attributeValueSetId',
             ],
 
             [
                 $languageMutator,
-                $skuMutator,
-                $defaultCategoryMutator
+                $defaultCategoryMutator,
+				$barcodeMutator
```
Comment (alinmigea) [inline src/ResultField/PreisRoboterDE.php]: add please here, the keyMutator

File: meta/documents/changelog_de.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Release Notes für Elastic Export PreisRoboter.de
 
+## v1.0.8 (2017-08-17)
```
Comment (alinmigea) [inline meta/documents/changelog_de.md]: maybe change the date

File: src/Generator/PreisRoboterDE.php [hunk 3]
```diff
@@ -39,117 +52,161 @@ public function __construct(ArrayHelper $arrayHelper)
...
+
+		$price = $priceList['price'];
+		$currency = '';
+
+		if((float)$price > 0)
+		{
+			$currency = $priceList['currency'];
+		}
+
+		$deliveryCost = $this->elasticExportHelper->getShippingCost($variation['data']['item']['id'], $settings);
```
Comment (alinmigea) [inline src/Generator/PreisRoboterDE.php]: maybe you could try to put a cache on the deliveryCost

File: src/Generator/PreisRoboterDE.php [hunk 4]
```diff
@@ -39,117 +52,161 @@ public function __construct(ArrayHelper $arrayHelper)
...
+
+				foreach($resultList['documents'] as $variation)
+				{
+					if($lines == $filter['limit'])
+					{
+						$limitReached = true;
+						break;
+					}
+
+					if(is_array($resultList['documents']) && count($resultList['documents']) > 0)
```
Comment (alinmigea) [inline src/Generator/PreisRoboterDE.php]: I think that this check is needed before the `foreach() `

**Review by alinmigea — CHANGES_REQUESTED** [review]

**Review by alinmigea — APPROVED** [review]

**Review by hschwab — APPROVED** [review]

**Pull request merged**